server/resp/src/main/java/org/infinispan/server/resp/commands/list/blocking/BPOP.java

```diff
@@ -0,0 +1,264 @@
+package org.infinispan.server.resp.commands.list.blocking;
+
+import java.lang.invoke.MethodHandles;
+import java.time.Duration;
+import java.util.ArrayDeque;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.function.BiConsumer;
+
+import org.infinispan.AdvancedCache;
+import org.infinispan.commons.logging.LogFactory;
+import org.infinispan.commons.marshall.WrappedByteArray;
+import org.infinispan.encoding.DataConversion;
+import org.infinispan.multimap.impl.EmbeddedMultimapListCache;
+import org.infinispan.multimap.impl.ListBucket;
+import org.infinispan.notifications.Listener;
+import org.infinispan.notifications.cachelistener.annotation.CacheEntryCreated;
+import org.infinispan.notifications.cachelistener.annotation.CacheEntryModified;
+import org.infinispan.notifications.cachelistener.event.CacheEntryEvent;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespErrorUtil;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.ArgumentUtils;
+import org.infinispan.server.resp.commands.Resp3Command;
+import org.infinispan.server.resp.filter.EventListenerConverter;
+import org.infinispan.server.resp.filter.EventListenerKeysFilter;
+import org.infinispan.server.resp.logging.Log;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * @link https://redis.io/commands/blpop/
+ *       Derogating to the above documentation, when multiple client are blocked
+ *       on a BLPOP, the order in which they will be served is unspecified.
+ * @since 15.0
+ */
+public class BPOP extends RespCommand implements Resp3Command {
```

Comment:
Looking at the list of blocking commands to add (ISPN-14625), do we abstract this in steps? I think we would need to abstract the parsing of arguments and the operation to execute.

Reply:
Yeah I created this PR just for BRPOP, since it's easy to implement over BLPOP with very trivial changes.
I would add abstraction in the next step, since other blocking commands may require indeep changes and I still need to look into.